node/build.gradle

```diff
@@ -17,58 +15,89 @@ def services = project.ext.services as List<Service>
 Map<String, String> serviceNaming = project.ext.serviceNamingCamel
 
 services.each { Service svc ->
+    // service name starting lowercase
     String serviceName = serviceNaming[svc.id]
+    // service unmodified (starting with uppercase)
+    String originalServiceName = svc.name
 
     // Generation
     tasks.named("generate${svc.name}", GenerateTask) {
-        apiPackage.set(serviceName)
-        configFile.set("$projectDir/config.yaml")
+        templateDir.set("$projectDir/repo/templates-v7/typescript")
+
+        apiNameSuffix.set('Api')
         additionalProperties.putAll([
-                'modelPropertyNaming': 'original',
-                'serviceName'        : serviceName,
+                'modelPropertyNaming'  : 'original',
+                'serviceName'          : serviceName,
+                'originalServiceName'  : originalServiceName,
+                'enumPropertyNaming'   : 'PascalCase',
         ])
+
+        // for webhooks apply extra config.yaml (to generate WebhookHandler)
+        if (serviceId.endsWith("webhooks")) {
+            configFile.set("$projectDir/config.yaml")
+        }
     }
 
-    // Deployment
+    // Copy models
     def deployModels = tasks.register("deploy${svc.name}Models", Sync) {
         group 'deploy'
         description "Deploy $svc.name models into the repo."
         dependsOn "generate$svc.name"
         outputs.upToDateWhen { false }
 
-        from layout.buildDirectory.dir("services/$svc.id/model")
+        from(layout.buildDirectory.dir("services/$svc.id/models")) {
+            eachFile { fileCopyDetails ->
+                def name = fileCopyDetails.name
+                if (name && name.length() > 0) {
+                    // copy and rename to start with lowercase
+                    fileCopyDetails.name = name[0].toLowerCase() + name.substring(1)
+                }
+            }
+
+            // Rename specific files
+            rename { fileName ->
+                if (fileName == "all.ts") {
+                    return "models.ts"
+                } else if (fileName == "webhookHandler.ts") {
+                    // rename to specific handler (i.e. reportWebhooksHandler.ts)
+                    def handlerName = "${svc.name}Handler.ts"
+                    return handlerName[0].toLowerCase() + handlerName.substring(1)
+                }
+                return fileName
+            }
+
+            includeEmptyDirs = false
+        }
         into layout.projectDirectory.dir("repo/src/typings/" + serviceName)
     }
 
+    // copy services
     def deployServices = tasks.register("deploy${svc.name}Services", Sync) {
         group 'deploy'
         description "Deploy $svc.name into the repo."
         dependsOn "generate$svc.name"
         outputs.upToDateWhen { false }
-        onlyIf { !svc.small && !svc.webhook }
-
-        from layout.buildDirectory.dir("services/$svc.id/${serviceName}")
-        include '*Api.ts'
+        onlyIf { !svc.webhook }
```

Comment:
Q: Explain the condition `onlyIf { !svc.webhook }` (why should this not be copied)